projects/giflib/egif_fuzz_common.cc

```diff
@@ -1,26 +1,36 @@
 #include "egif_fuzz_common.h"
-#define GIF_IMAGE_WIDTH  100
+#define GIF_IMAGE_WIDTH 100
 // This is rgb byte stream length per horizontal line = GIF_IMAGE_WIDTH * 3
-#define GIF_IMAGE_LINE   300
+#define GIF_IMAGE_LINE 300
 
 extern "C" void PrintGifError(int ErrorCode);
 
-int stub_output_writer(GifFileType* gifFileType, const uint8_t* buf, int len)
+int stub_output_writer(GifFileType *gifFileType, const uint8_t *buf, int len)
 {
-	struct gifUserData* gud = (struct gifUserData*) gifFileType->UserData;
+	struct gifUserData *gud = (struct gifUserData *)gifFileType->UserData;
 
 	if (gud == NULL || gud->gifData == NULL || len == 0)
 		return 0;
-
-	memcpy(gud->gifData, buf, len);
-	gud->gifData += len;
+	if (gud->allocatedSize < (gud->gifLen + len))
+	{
+		// Reallocate gifFileType
+		int newSize = gud->allocatedSize * 2;
```

Comment:
If `newSize` is still less than `gud->gifLen + len`, buffer over runs.

Reply:
Whops, really good catch! What is the best way to deal with that:
std::max(newSize, gud->gifLen + len);?

Reply:
perhaps `int newSize = (gud->gifLen + len) * 2;`?

Reply:
sounds good

Reply:
Fixed :)